### ISSUE TITLE
doxygen syntax highlight error with @param[in]

### ISSUE THREAD

**Issue by Gorgeous2017** [microsoft/vscode#90065]
Issue Type: <b>Bug</b>

When use the doxygen tag @param[in/out], the variable could highlihght normally.

My code such as:
```c
/**
 * @brief xxx
 * 
 * @param [IN] par1 xxxx
 */
void Function_1(void par1) {
xxx
}
```

When I add '[IN]' after the '@param',  'par1' could highlight. 

When I change the format as '@param[IN]', '@param[in]', '@param [in]', 'par1' couldn't highlight as well.

Am I mistaken the usage of doxygen? 

VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:58:56.166Z)
OS version: Windows_NT x64 10.0.18362

**Comment by alexr00** [comment]
Please also provide a screen shot of what you're seeing and what you expect to see.

**Comment by Gorgeous2017** [comment]
> Please also provide a screen shot of what you're seeing and what you expect to see.

![image](https://user-images.githubusercontent.com/33201519/73924363-5782a580-4907-11ea-804a-be3a9e504084.png)

Okay. This is what I seeing.

I expect to see 'par1' wihch in  Unnormal Condition change the color same as in Normal Condition.

**Comment by alexr00** [comment]
I don't see anything in [the manual](http://doxygen.nl/manual/docblocks.html) about Doxygen supporting IN instead of in. If you can point me to somewhere this says it's supported then I'll pass the bug on to the owners of the C++ syntax highlighting grammar.

**Comment by Gorgeous2017** [comment]
[Doxygen Manual: Special Commands](http://www.doxygen.nl/manual/commands.html#cmdparam)

Please see Doxygen Manual - Special Commands - \param '['dir']' <parameter-name> { parameter description }

**Comment by alexr00** [comment]
Ah my bad, I tested running out of source and I just pulled in the latest C++ grammar this week. This is fixed in master and you'll start seeing the fix in insiders with version 1.43.

**Comment by Gorgeous2017** [comment]
I find a same issue in jeff-hykin/cpp-textmate-grammar#418, and it was fix by jeff-hykin/cpp-textmate-grammar#419 PR which merge to master at 15 days ago. Hope this information is useful for you.

**Comment by Gorgeous2017** [comment]
@alexr00 
hey, I noticed that this bug is fixed in v1.43.0. But today I update to v1.43.1, this bug **happend again**.....What wrong with it??

**Comment by alexr00** [comment]
The version of the C++ grammar we used in 1.43.0 caused a freeze, so for 1.43.1 we reverted back to the version without the freeze. If the freeze is fixed in the C++ grammar, then I'll update it again and you'll get this fix back. In the meantime, we get our grammar from https://github.com/jeff-hykin/cpp-textmate-grammer. You can get this grammar yourself by installing the Better C++ Syntax extension.
